types/genesis_test.go

```diff
@@ -14,14 +14,10 @@ import (
 func TestGenesisBad(t *testing.T) {
 	// test some bad ones from raw json
 	testCases := [][]byte{
-		[]byte{},                                           // empty
-		[]byte{1, 1, 1, 1, 1},                              // junk
-		[]byte(`{}`),                                       // empty
-		[]byte(`{"chain_id":"mychain"}`),                   // missing validators
-		[]byte(`{"chain_id":"mychain","validators":[]}`),   // missing validators
```

Comment:
can we not remove these, but move them to TestGenesisGood instead?

Reply:
They supposed to be bad genesis json? shall I move them to `GoodGenesisTest`?
I added new test with wrong public key.

Reply:
Done. Please check the latest commit.

Reply:
Oh god. I think you misunderstood me. I wanted you to only move "missing validators" lines, not the whole `TestGenesisBad` test. The reason for moving is that "missing validators" are now correct use-case, so imho should be inside `TestGenesisGood`

Reply:
Please check it now